### DISCUSSION TITLE
How can i calculate transaction fees for bridge transactions?

### DISCUSSION THREAD

**Discussion by AmadiMichael** [0xPolygon/matic.js#352] [Q&A]
Is there a way that i can calculate the tx fees a bridge swap will cost before hand? 
Ethers "estimateGas" doesn't seem to work on it

Polygon wallet displays it so there must be a way. 
anyone got a way around this?

A way to get the gasLimit before hand to be precise?

**Comment by ujjwalgupta94** [comment]
yeah you can calculate gas Limit using `estimateGas` method similar to how you do it in ethereum. You can take a look at maticjs ethers repo for ether.js handling - https://github.com/maticnetwork/maticjs-ethers/blob/70a0f5d3fd91118038b9afd7d73278bd0599a34e/src/ethers/web3_client.ts#L95

**Reply by AmadiMichael** [reply]
Thanks let me try it out.

**Reply by ujjwalgupta94** [reply]
if you are using maticjs, you don't need to worry about estimateGas, as its already calculate this and add in every transaction.

**Reply by AmadiMichael** [reply]
I'm looking for a way to get the details before hand and display it.

**Comment by AmadiMichael** [comment]
> yeah you can calculate gas Limit using `estimateGas` method similar to how you do it in ethereum. You can take a look at maticjs ethers repo for ether.js handling - https://github.com/maticnetwork/maticjs-ethers/blob/70a0f5d3fd91118038b9afd7d73278bd0599a34e/src/ethers/web3_client.ts#L95

How do I use it. Please can you give an example?
